### ISSUE TITLE
[FR] Add zstd support for MLflow model compression

### ISSUE THREAD

**Issue by punkeel** [mlflow/mlflow#25639]
> [!WARNING]
> Before submitting a PR, please make sure that:
> - A maintainer has triaged this issue and applied the `ready` label
> - This issue has no assignee
> - No duplicate PR exists
>
> PRs not meeting these requirements may be automatically closed.

### Willingness to contribute

Yes. I would be willing to contribute an implementation, either directly to MLflow or as a plugin, after agreeing on the API and compatibility approach with maintainers.

### Proposal Summary

Add Zstandard (`zstd`) as a supported compression method for MLflow model logging and checkpoint artifacts, extending the compression support introduced in #9033 and implemented for `pyfunc` in #14700.

Users should be able to select zstd through the existing model-compression configuration mechanism, for example:

```bash
MLFLOW_LOG_MODEL_COMPRESSION=zstd
```

The compressed artifact should remain discoverable and downloadable through normal MLflow APIs, with decompression handled transparently where practical.

### Motivation

Modern training workflows may produce many large model checkpoints. Retaining every checkpoint is valuable for resuming interrupted training, analyzing model behavior across training steps, comparing intermediate checkpoints, reproducing results, and auditing training decisions.

Uncompressed model artifacts can consume substantial disk and object-store capacity. Users currently need to implement their own zstd compression and upload conventions, which reduces interoperability and makes artifact handling less consistent.

Zstandard provides a strong speed-to-compression tradeoff and is widely available across languages and platforms. Native support would make it practical to retain complete checkpoint histories without discarding useful artifacts or building custom wrappers around MLflow.

This is especially valuable for long-running deep-learning training, hyperparameter sweeps, checkpoint evaluation, and recovery from preempted jobs.

### Details

The proposed implementation could:

1. Add `zstd` as an accepted compression method for model logging, alongside the existing methods.
2. Compress model files before artifact upload using a maintained zstd implementation.
3. Use an unambiguous compression suffix such as `.zst` and preserve the original model format in metadata.
4. Decompress transparently during model/artifact download where MLflow controls the loading path.
5. Preserve compatibility with existing uncompressed, gzip, bzip2, and lzma artifacts.
6. Expose the selected compression method in run/model metadata.
7. Provide clear errors when zstd support is unavailable.
8. Add tests for compression, upload/download, loading, corrupted artifacts, and backward compatibility.
9. Document compression-level configuration if maintainers consider it useful.

A follow-up or complementary enhancement could extend the same compression mechanism to generic checkpoint artifacts logged through `log_artifact` / `log_artifacts`, while keeping ordinary plots, reports, and metadata unchanged.

The implementation should avoid requiring users to retain both compressed and uncompressed copies after upload, and should not change the logical model format presented to callers.

### What machine learning domain(s) is this feature request about?

`domain/platform`

`domain/deep-learning`

### What area(s) of MLflow is this feature request about?

`area/tracking`

`area/models`
